### PR TITLE
Fix typespec since password may be a string() or a fun()

### PR DESCRIPTION
**Pull request by onno-vos-dev**
Dialyzer is not so happy due to the wrong spec :)

### CODE REVIEW

File: src/epgsql.erl [hunk 1]
```diff
@@ -44,11 +44,12 @@
 
 -type sql_query() :: iodata().
 -type host() :: inet:ip_address() | inet:hostname().
+-type password() :: string() | fun().
```
Comment (seriyps) [inline src/epgsql.erl]: Can you, please, define `fun()` as `fun( () -> iodata() )`? Also, password can be not only a string but binary or any iodata as well

Comment (onno-vos-kivra) [reply src/epgsql.erl]: Updated. Since `iodata()` is `iolist() | binary()` I did not add `binary()` explicitly.

Comment (seriyps) [reply src/epgsql.erl]: @onno-vos-kivra perfect, thanks!

**Comment by seriyps** [comment]
Thank you! :+1:

**Pull request merged**